contracts/PepemonCardDeck.sol

```diff
@@ -77,40 +85,84 @@ contract PepemonCardDeck is ERC721, ERC1155Holder, Ownable {
     }
 
-    function setMinSupportCards(uint256 _minSupportCards) public onlyOwner {
+    function setMinSupportCards(uint256 _minSupportCards) public onlyAdmin {
         MIN_SUPPORT_CARDS = _minSupportCards;
     }
 
+    function setRandNrGenContractAddress(address randOracleAddress) public onlyAdmin {
+        randNrGenContract = randOracleAddress;
+    }
+
+    function setInitialDeckOptions(
+        uint256[] calldata battleCardIds,
+        uint256[] calldata supportCards,
+        uint256 maxInitialSupportCards
+    ) public onlyAdmin {
+        require(Arrays.isSortedAscending(battleCardIds));
+
+        initialDeckSupportCardAmount = maxInitialSupportCards;
+        allowedInitialDeckBattleCards = battleCardIds;
+        allowedInitialDeckSupportCards = supportCards;
+    }
+
     // ALLOW TEST MINTING
-    function setMintingCards(uint256 minCardId, uint256 maxCardId) public onlyOwner {
+    function setMintingCards(uint256 minCardId, uint256 maxCardId) public onlyAdmin {
         maxMintTestCardId = maxCardId;
         minMintTestCardId = minCardId;
     }
+
     /**
-     * @dev right now there are 40 different cards that can be minted, but the maximum is configurable with maxMintTestCard. 
+     * @dev right now there are 40 different cards that can be minted, but the maximum is configurable with maxMintTestCard.
      * setting maxMintTestCard to 0 disables this card minting.
      */
     function mintCards() public {
         require(maxMintTestCardId > 0, "Minting test cards is disabled");
         IPepemonFactory(supportCardAddress).batchMint(minMintTestCardId, maxMintTestCardId, msg.sender);
     }
 
+    function mintInitialDeck(uint256 battleCardId) public {
+        require(Arrays.contains(allowedInitialDeckBattleCards, battleCardId), "Invalid battlecard");
+        require(playerToDecks[msg.sender].length == 0, "Not your first deck");
+        // battlecard + support cards
+        uint amount = initialDeckSupportCardAmount + 1;
+        uint256[] memory cards = new uint256[](amount);
+
+        // First step: Mint cards
+
+        uint256 allowedCardsCount = allowedInitialDeckSupportCards.length;
+        uint256 randomNumber = randSeed();
+        cards[0] = battleCardId;
+        // begin from index 1 instead of 0 because battlecard was already added
+        for (uint256 i = 1; i < amount; ++i) {
+            randomNumber = uint256(keccak256(abi.encodePacked(i, randomNumber)));
+            cards[i] = allowedInitialDeckSupportCards[randomNumber % allowedCardsCount];
+        }
+        IPepemonFactory(battleCardAddress).batchMintList(cards, msg.sender);
+    
+        // Second step: Add cards into new the deck
+        
+        uint256 newDeckId = createDeckInternal();
+        addBattleCardToDeck(newDeckId, battleCardId);
+        // begin from index 1 again because battlecard is in index 0
+        for (uint256 i = 1; i < amount; ++i) {
+            addSupportCardToDeck(newDeckId,cards[i], 1);
+        }
```

Comment:
This mints the cards to `msg.sender` then transfers them back to this contract. It will require approval for cards the player doesn't have yet, which might be problematic for some wallets.

Could we mint as owned by this contract, then update the deck directly? (no transfers)